sapl/relatorios/forms.py

```diff
@@ -68,6 +70,73 @@ def __init__(self, *args, **kwargs):
         )
 
 
+class RelatorioVotacoesNominaisFilterSet(django_filters.FilterSet):
+
+    tipo_id = django_filters.ModelChoiceFilter(
+            queryset=TipoMateriaLegislativa.objects.all(),
+            method='ordem_or_expediente',
+            label='Tipo de Matéria',
+            empty_label="---------"
+        )
+    numero = django_filters.NumberFilter(
+            widget=forms.NumberInput(attrs={'class': 'form-control', 'step': 'any'}),
+            method='ordem_or_expediente',
+            label='Número'
+        )
+    ano = django_filters.ChoiceFilter(
+        choices=list(choice_anos_com_materias()),
+        widget=forms.Select(attrs={'class': 'form-control'}),
+        method='ordem_or_expediente',
+        label='Ano da Matéria'
+        )
+
+    def ordem_or_expediente(self, queryset, name, value):
+        if value:
+            try:
+                val = value.id
+            except AttributeError:
+                val = value
+            return queryset.filter(
+                    eval(f'Q(ordem__materia__{name}={val}) | Q(expediente__materia__{name}={val})')
+                   )
```

Comment:
Isso me parece arriscado, e um code smell... avaliar um valor externo para construção de um parâmetro de query **em tempo de execução** não parece elegante, nem correto e abre brecha pra ataques de segurança...


```suggestion
            ordem_q = f"ordem__materia__{name}"
            expediente_q = f"expediente__materia__{name}"
            return queryset.filter(Q(**{ordem_q: val})|Q(**{expediente_q: val}))
```